smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelParser.java

```diff
@@ -937,6 +938,18 @@ private void parseResourceStatement(ShapeId id, SourceLocation location) {
                 modelFile.addForwardReference(target.getValue(), targetId -> builder.addIdentifier(name, targetId));
             }
         });
+        // Load properties and resolve forward references.
+        shapeNode.getObjectMember(PROPERTIES_KEY).ifPresent(properties -> {
+            if (!modelFile.getVersion().supportsResourceProperties()) {
```

Comment:
Need a similar check in AstModelLoader. Maybe move it to LoaderUtils?

Reply:
I added both checks. Can't use LoaderUtils sensibly due to both checks having different flow-control reactions.